lib/colorado_lottery.rb

```diff
@@ -0,0 +1,28 @@
+class ColoradoLottery
+  attr_reader :registered_contestants,
+              :winners,
+              :current_contestants
+
+  def initialize
+    @registered_contestants = Hash.new
+    @winners = []
+    @current_contestants = []
+  end
+
+  def interested_and_18?(contestant, game)
+   contestant.game_interests.include?(game.name) && contestant.age >= 18
+  end
+
+  def can_register?(contestant, game)
+    interested_and_18?(contestant, game) && (!contestant.out_of_state? || game.national_drawing?)
+  end
+
+  def register_contestant(contestant, game)
+    (@registered_contestants[game.name] ||= []) << contestant
```

Comment:
This doesn't work because the return value of the code in the parens is an array; not **the** array that is associated with the key.  This would work:

```ruby
@registered_contestants[game.name] = [] if @registered_contestants[game.name].nil?
@registered_contestants[game.name] << contestant
```